x-pack/plugins/security/server/authentication/index.ts

```diff
@@ -75,8 +86,15 @@ export async function setupAuthentication({
       encryptionKey: config.encryptionKey,
       isSecure: config.secureCookies,
       name: config.cookieName,
-      validate: (sessionValue: ProviderSession) =>
-        !(sessionValue.expires && sessionValue.expires < Date.now()),
+      validate: (session: ProviderSession) => {
+        const array: ProviderSession[] = Array.isArray(session) ? session : [session];
+        for (const sess of array) {
+          if (!isValid(sess)) {
+            return { isValid: false, path: sess.path };
```

Comment:
so, validate returns on the very first invalid cookie and remove only it?

Reply:
Yes, that's correct.

It's because we're using the Hapi toolkit to clear the cookie (which is the same method that the Hapi Cookie plugin uses under the hood in our current implementation -- we are just passing it an extra parameter). However, the Hapi toolkit can only clear one cookie at a time. This sounded like an acceptable approach to me because:
* The only time we'll ever see multiple cookies is if the client has stored several more than one cookie with the same name and different, overlapping paths (e.g., `/` and `/foo`)
* If multiple cookies are sent and they are all invalid (very unlikely), the client will make multiple round trips and delete each one; this will happen near-instantaneously and be imperceptible to the end user

The alternative was to try to reimplement that logic in a different way so we could delete multiple cookies in a single response, but that seemed unnecessary and risky -- we know the existing method works, and we want to avoid mucking around with Hapi under the hood as much as possible.